### PR TITLE
ResourceSerializer: Determine price type based on price list (TTVA-106)

### PR DESCRIPTION
**Pull request by jopesy**
If the resource has a price list and the price list has prices other than just fixed prices, return `mixed` as the price type so the client knows the price information needs to be displayed differently.

Refs TTVA-106

### CODE REVIEW

File: respa_pricing/models/price.py [hunk 1]
```diff
@@ -120,6 +121,25 @@ class Meta:
     def __str__(self):
         return self.name
 
+    @property
```
Comment (danjacob-anders) [inline respa_pricing/models/price.py]: Couple points:

1. Assuming not likely to change, would a `cached_property` make more sense to reduce re-running db queries?
2. As we are fetching identical fields from different models, maybe `union()` would be a bit more efficient:

```python
price_types = set(
   self.usergroup_prices
   .union(self.event_prices.all()
)).values_list("price_type", flat=True).distinct())
```

Comment (jopesy) [reply respa_pricing/models/price.py]: Both are good suggestions. I'll make the changes.

Comment (jopesy) [reply respa_pricing/models/price.py]: Modified to use `cached_property` and `union`. It's much much better now. Thanks for the comment!

**Review by danjacob-anders — APPROVED** [review]
Looks good!

**Pull request merged**